src/routes/Apps/App/TrainDialogs.tsx

```diff
@@ -922,13 +951,18 @@ class TrainDialogs extends React.Component<Props, ComponentState> {
 
     @OF.autobind
     async openTrainDialog(trainDialog: CLM.TrainDialog, roundIndex: number, scoreIndex: number | null) {
-
         const selectedActivityIndex = DialogUtils.activityIndexFromRound(trainDialog, roundIndex, scoreIndex) || null
-        await this.onClickTrainDialogItem(trainDialog, EditDialogType.TRAIN_ORIGINAL, selectedActivityIndex)
+        await this.selectTrainDialog(trainDialog, EditDialogType.TRAIN_ORIGINAL, selectedActivityIndex)
     }
 
     @OF.autobind
-    async onClickTrainDialogItem(trainDialog: CLM.TrainDialog, editType: EditDialogType = EditDialogType.TRAIN_ORIGINAL, selectedActivityIndex: number | null = null) {
+    async onClickTrainDialogItem(trainDialog: CLM.TrainDialog) {
+        const { history } = this.props
+        let url = `/home/${this.props.app.appId}/trainDialogs?id=${trainDialog.trainDialogId}`
```

Comment:
Should be a way to use `match.url` here. Which is the current state of router. This way we can change parent paths such as home or trainDialogs an this will still work instead of being hard coded.

Reply:
Again, would use the constant mentioned above when generating the querystring so it can't get mismatched. Optional since it's simple case but generating the query string using one of the many libraries. We already depend on 3 different packages (queryString, querystring, query-string) which do the same thing so might as well use them.

Putting it all together I was envisioning something like this:
```
import * as queryString from 'query-string'
...
const queryObject = {
  [queryParams.id]: trainDialog.trainDialogId
}

const query = queryString.stringify(queryObject)

${match.url}?${query}
```